tests/src/test_lambda_alerting.py

```diff
@@ -0,0 +1,325 @@
+    )
+
+@pytest.fixture(scope="session")
+def event_dyn_props_init(aws_props_init):
+     # Generates dynamic properties for events
+    (account_id, region) = aws_props_init
+    current_time = datetime.now() - timedelta(minutes=1)
+    epoch_time = int(current_time.timestamp())
+    time_str = current_time.strftime("%Y-%m-%dT%H:%M:%SZ")
+    version_str = current_time.strftime("%Y%m%d%H%M%S")
+
+    return (account_id, region, time_str, epoch_time, version_str)    
+
+################################################################################################################################
+
+# GLUE JOB tests 
+
+# Utility function to generate a Glue Job event with dynamic properties.
+def get_glue_job_event(event_dyn_props, state, message):
+    (account_id, region, time_str, _, version_str) = event_dyn_props
+
+    return {
+        "version": version_str, # instead of "normal" '0', we put here custom value, so we can track and find event in CloudWatch LogInsights
+        "id": "abcdef00-1234-5678-9abc-def012345678",
+        "detail-type": "Glue Job State Change",
+        "source": "aws.glue",
+        "account": account_id,
+        "time": time_str,
+        "region": region,
+        "resources": [],
+        "detail": {
+            "jobName": "glue-salmonts-pyjob-1-dev",
+            "severity": "INFO",
+            "state": state, #"SUCCEEDED"/"FAILED",
+            "jobRunId": "jr_abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789",
+            "message": message, #"Job run succeeded", etc,
+        },
+    }
+
+def test_glue_job1(os_vars_init, event_dyn_props_init):
+    
+    event = get_glue_job_event(event_dyn_props_init, "FAILED", "Job run failed")
+
+    result = lambda_handler(event, {})
+
+    assert result["event_is_alertable"] == True, "Event should raise alert"
+    assert result["event_is_monitorable"] == True, "Event should be logged"
+    assert (
+        result["resource_type"] == resource_types.GLUE_JOBS
+    ), "Event should be monitorable"
+
+def test_glue_job2(os_vars_init, event_dyn_props_init):
+    
+    event = get_glue_job_event(event_dyn_props_init, "SUCCEEDED", "Job run succeeded")
+
+    result = lambda_handler(event, {})
+
+    assert result["event_is_alertable"] == False, "Event shouldn't raise alert"
+    assert result["event_is_monitorable"] == True, "Event should be logged"
+    assert (
+        result["resource_type"] == resource_types.GLUE_JOBS
+    ), "Event should be monitorable"    
+
+def test_glue_job3(os_vars_init, event_dyn_props_init):
+    
+    event = get_glue_job_event(event_dyn_props_init, "RUNNING", "Job is running")
+
+    result = lambda_handler(event, {})
+
+    assert result["event_is_alertable"] == False, "Event shouldn't raise alert"
+    assert result["event_is_monitorable"] == False, "Event shouldn't be logged"
+    assert (
+        result["resource_type"] == resource_types.GLUE_JOBS
+    ), "Event should be monitorable"      
```

Comment:
Not sure I understand the meaning of the comment. event_is_monitorable == False, however we comment that "Event should be monitorable", while it's just a resource type check.

Reply:
Thanks!
That's a wicked power of copy-and-paste :)
Fixed